src/core/src/op/loop.cpp

```diff
@@ -191,6 +191,57 @@ void op::v5::Loop::validate_and_infer_types() {
     // Body
     m_bodies[0]->validate_nodes_and_infer_types();
 
+    // if output shape is not same with input in a back-edge, here will update the input shape
+    // Port map processing: output -> input
+    std::map<uint64_t, uint64_t> back_edges;
+    for (const auto& desc : get_input_descriptions()) {
+        if (const auto& merged_desc = std::dynamic_pointer_cast<opset5::Loop::MergedInputDescription>(desc)) {
+            back_edges[merged_desc->m_body_value_index] = merged_desc->m_body_parameter_index;
+        }
+    }
+    if (!back_edges.empty()) {
+        bool need_reinvalidate = false;
+        for (const auto& output_description : m_output_descriptions[0]) {
+            auto body_value = m_bodies[0]->get_results().at(output_description->m_body_value_index)->input_value(0);
+
+            if (auto body_output_description =
+                    ov::as_type_ptr<v0::TensorIterator::BodyOutputDescription>(output_description)) {
+                const ov::PartialShape& body_value_shape = body_value.get_partial_shape();
+                if (body_value_shape.rank().is_static()) {
+                    // handle the case: when sub-model's output shape does not compatible input shape in a back-edge,
```

Comment:
Should we handle also scalar (Shape{}) and Shape{1} here in special case or it is already covered?
If not, maybe it is a topic for separate PR.

Reply:
Add special case {}<=>{1} support.